docs/changelog/v6.md

```diff
@@ -0,0 +1,214 @@
+
+Most Halogen 5 applications solve this problem by rendering these components at the root of the application, and using messages or a global state to control when they render. For example, if a small widget needs to render an error notification, you might push to a queue in global state, which the root component then uses to render the appropriate notification on top of all other content.
+
+Portals make this state management and threading unnecessary by allowing you to render a HTML or a child component to any node in the DOM. That small widget can now render the error notification directly to the `<body>` or a notifications-specific div outside the main application tree.
+
+The `portal` function takes a target element and a `ComponentHTML` and renders the HTML within the target element. You can take any existing `ComponentHTML` and simply provide it to `portal`. For example:
+
+```purs
+HH.portal targetElement do
+  HH.div ...
+
+HH.portal targetElement do
+  HH.slot ...
+```
+
+Added in https://github.com/purescript-halogen/purescript-halogen/pull/705.
+
+## Component Types
+
+Component types have been simplified by removing the `surface` parameter.
+
+In Halogen 5 (and prior versions), components and the internal functions which manage them carried a `surface` parameter which indicated the target for the UI to render. As no one ever wrote an alternate target from `HTML`, Halogen applications have always fixed this parameter to `HTML` in component definitions, as in:
+
+```purs
+import Halogen as H
+import Halogen.HTML as HH
+
+myComponent :: forall q i o m. H.Component HH.HTML q i o m
+```
+
+In Halogen 6 the `surface` parameter has been removed. The only real user-visible change is that components and functions which operate on them no longer carry the `surface` parameter.
+
+```purs
+import Halogen as H
+
+myComponent :: forall q i o m. H.Component q i o m
+```
+
+This is a breaking change, but one which is easily fixed: remove this parameter from your components and any related functions and types.
+
+Added in [#616](https://github.com/purescript-halogen/purescript-halogen/pull/616).
+
+## Event Handler Types
+
+We've also made event handlers a little nicer to work with. The `Maybe action` return value has been removed from event handlers, which now return `action` directly.
+
+In Halogen 5, when you wanted to respond to a click event, or a message from a child component, the output was of type `Maybe action`. This allowed you to selectively emit outputs (you could emit `Nothing` for some events). In practice, though, almost no one ever does this.
```

Comment:
Would you like to add a note that users can add a `DoNothing` constructor to their action type or replace the action type with `Maybe action` instead, if they need to keep the old behavior?

Reply:
Added!